pkg/resource/deploy/source_eval.go

```diff
@@ -959,6 +958,44 @@ func (rm *resmon) Invoke(ctx context.Context, req *pulumirpc.ResourceInvokeReque
 		return nil, fmt.Errorf("failed to unmarshal %v args: %w", tok, err)
 	}
 
+	opts := &pulumirpc.TransformInvokeOptions{
+		Provider:          req.GetProvider(),
+		Version:           req.GetVersion(),
+		PluginDownloadUrl: req.GetPluginDownloadURL(),
+		PluginChecksums:   req.GetPluginChecksums(),
+	}
+
+	// Then lock the stack transformations and run all of those
```

Comment:
Depending on the previous comment, this one might also need tweaking.

Reply:
In fact we should probably call out that we're running the transforms first precisely because they may influence the provider load (assuming that's right).